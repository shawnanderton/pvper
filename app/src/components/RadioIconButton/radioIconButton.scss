$selected-color: #e02929;

.radio-icon-button {
  padding: 0;
  margin: 0;
  border-radius: 3px;
  border: none;
  background-color: inherit;
}

.selected {
  border: solid 1px $selected-color;
}
