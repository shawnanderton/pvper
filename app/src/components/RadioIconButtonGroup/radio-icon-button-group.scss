.radio-icon-button-group {
  margin-left: 10px;
  .items-group {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    flex-direction: row;
    border-color: #212121;

    .items {
      display: flex;
      flex-wrap: wrap;
      justify-content: space-evenly;
      align-content: space-between;

      .item{
        margin: 0 10px 5px 0;
      }
      
      
  
    }
    
  }
}
