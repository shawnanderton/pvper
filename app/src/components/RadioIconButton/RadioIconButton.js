import React, { useState } from 'react';

import Icon from '../Icon';

import './radioIconButton.scss';

function RadioIconButton({
  id,
  selected = false,
  icon = { name: '', size: 24 },
  onChange,
}) {
  const handleClick = () => {
      console.log(selected);
    onChange({ key: id, selected: !selected });
  };
  return (
    <button
      onClick={handleClick}
      className={`radio-icon-button ${selected ? 'selected' : ''}`}
    >
      <Icon name={icon.name} size={icon.size} />
    </button>
  );
}

export default RadioIconButton;
