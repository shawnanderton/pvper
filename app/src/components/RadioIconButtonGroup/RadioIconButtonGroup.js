import React from 'react';
import RadioIconButton from '../RadioIconButton';

import './radio-icon-button-group.scss';

function RadioIconButtonGroup({ label, items, onChange }) {
  function handleChange(item) {
    const index = items.findIndex((i) => i.key === item.key);
    items[index].selected = item.selected;
    onChange(items);
  }

  function selectAll(event) {
    event.preventDefault();
    onChange(items.map((i) => ({ ...i, selected: true })));
  }

  return (
    <div className="radio-icon-button-group">
      <h4>{label}</h4>
      <div className="items-group">
        <div>
          <div className="items">
            {items.map((item) => (
              <div className="item">
                <RadioIconButton
                  onChange={handleChange}
                  key={item.key}
                  id={item.key}
                  selected={item.selected}
                  icon={item.icon}
                />
              </div>
            ))}
          </div>
        </div>

        <a href="#" onClick={selectAll} className="item-all">
          <strong>All</strong>
        </a>
      </div>
    </div>
  );
}

export default RadioIconButtonGroup;
