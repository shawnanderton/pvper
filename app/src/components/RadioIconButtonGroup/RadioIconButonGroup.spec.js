import React from 'react'
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

import RadioIconButtonGroup from './RadioIconButtonGroup';
import RadioIconButton from '../RadioIconButton';

describe('RadioIconButtonGroup', () => {
     it('should render correctly', () => {
    const tree = renderer
      .create(
        <RadioIconButtonGroup items={[]} />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
it('should render to  ', () => {
    const items = [
        { selected:true, icon:{name:'horde', size:24}}
    ]
    const wrapper = mount(<RadioIconButtonGroup items={items} />);
    console.log(wrapper.find("fieldset").debug());
    expect(wrapper.find("fieldset").children().length).toBe(1);
});

});