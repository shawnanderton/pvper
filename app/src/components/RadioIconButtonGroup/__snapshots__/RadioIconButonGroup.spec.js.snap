// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioIconButtonGroup should render correctly 1`] = `<fieldset />`;
